### PR TITLE
Cleanup sphinx examples

### PR DESCRIPTION
**Pull request by hjmjohnson**

### CODE REVIEW

File: src/Filtering/ImageGradient/ApplyGradientRecursiveGaussian/Code.py [hunk 1]
```diff
@@ -45,7 +45,7 @@
 FloatImageType = itk.Image[FloatPixelType, Dimension]
 # The output of GradientRecursiveGaussianImageFilter
 # are images of the gradient along X and Y, so the type of
-# the output is a covariant vector of dimension 2 (X, Y)
+# the output is a covariant vector of l_dimension 2 (X, Y)
```
Comment (dzenanz) [inline src/Filtering/ImageGradient/ApplyGradientRecursiveGaussian/Code.py:48]: is this change intentional?

File: src/Filtering/ImageGrid/ResampleAnImage/Code.py [hunk 2]
```diff
@@ -33,7 +33,7 @@
 
 # We will scale the objects in the image by the factor `scale`; that is they
 # will be shrunk (scale < 1.0) or enlarged (scale > 1.0).  However, the number
-# of pixels for each dimension of the output image will equal the corresponding
+# of pixels for each l_dimension of the output image will equal the corresponding
```
Comment (dzenanz) [inline src/Filtering/ImageGrid/ResampleAnImage/Code.py:36]: probably unwanted change

File: src/Core/Transform/GlobalRegistrationTwoImagesBSpline/Code.cxx [hunk 3]
```diff
@@ -42,9 +41,9 @@ using PixelType = float;
 using ImageType = itk::Image<PixelType, ImageDimension>;
 
 static void
-CreateEllipseImage(ImageType::Pointer image);
+CreateEllipseImage(const ImageType::Pointer & image);
```
Comment (dzenanz) [inline src/Core/Transform/GlobalRegistrationTwoImagesBSpline/Code.cxx:40]: We are modifying this image, so the `const` qualified is confusing.

**Review by dzenanz — COMMENTED** [review]
A good cleanup!

File: src/Core/Transform/GlobalRegistrationTwoImagesBSpline/Code.cxx (hunk 3 above)
Comment (hjmjohnson) [reply src/Core/Transform/GlobalRegistrationTwoImagesBSpline/Code.cxx:40]: We are not modifying the pointer though.

Comment (hjmjohnson) [reply src/Core/Transform/GlobalRegistrationTwoImagesBSpline/Code.cxx:40]: This was a clang-tidy warning.

**Review by dzenanz — APPROVED** [review]
